ui/slider.js

```diff
@@ -410,9 +410,24 @@ return $.widget( "ui.slider", $.ui.mouse, {
 		}
 	},
 
-	_setOption: function( key, value ) {
+	_changeAllValues: function( valuesLength ) {
 		var i,
-			valsLength = 0;
+			rangeValues = [ this.values( 0 ), this.values( 1 ), this._valueMax() ];
+
+		if ( this.options.range === true && $.unique( rangeValues ).length === 1 ) {
```

Comment:
Is this check actually necessary? Might be enough to just always do the reverse loop. I tested that locally with the new visual test page and the unit tests, works for both.